drivers/flash/soc_flash_nrf.c

```diff
@@ -513,6 +546,9 @@ static int write_op(void *context)
 	}
 #endif /* CONFIG_SOC_FLASH_NRF_RADIO_SYNC */
 
+	/* Enable write operation */
```

Comment:
f. flash_nrf_write_protection() is for enable write explicite. Pleas remove this form here - its breaks API compatibility.